Automattic/wp-calypso
# Switch Site: remove site icons from All Sites button

#### Proposed Changes

Remove the site icons from the All Sites button to reduce visual clutter. I don't think these added much value, so I experimented with removing them to make the site selector less noisy. This also more closely replicates the design mock-ups that @SaxonF shared for the project.

|Before|After|
|-|-|
|<img width="273" alt="Screenshot 2022-10-07 at 18 40 54" src="https://user-images.githubusercontent.com/1182160/194611350-40a49de8-048b-425f-b395-946181c80e55.png">|<img width="272" alt="Screenshot 2022-10-07 at 18 40 24" src="https://user-images.githubusercontent.com/1182160/194611386-2c77c58b-a5c1-4a9e-a6c3-4bfe657b021f.png">|


| Before hover | After hover |
|-|-|
|<img width="272" alt="Screenshot 2022-10-07 at 18 44 21" src="https://user-images.githubusercontent.com/1182160/194611610-f15d5ad7-e839-4a2d-a826-d5c128d4a9f0.png">|<img width="272" alt="Screenshot 2022-10-07 at 18 44 41" src="https://user-images.githubusercontent.com/1182160/194611630-5abe642c-985b-4c29-9d76-bc444ac38aa1.png">|

| Jetpack Cloud before | Jetpack Cloud After |
|-|-|
|<img width="273" alt="Screenshot 2022-10-07 at 19 21 03" src="https://user-images.githubusercontent.com/1182160/194615054-8c1081e8-f7e4-46f8-8f97-2a3a30e8de67.png">|<img width="272" alt="Screenshot 2022-10-07 at 19 20 41" src="https://user-images.githubusercontent.com/1182160/194615702-62e2ce52-fa2f-4d70-9acf-7bf055b15639.png">|
|<img width="273" alt="Screenshot 2022-10-07 at 19 21 39" src="https://user-images.githubusercontent.com/1182160/194615139-e92784d7-1943-4f5b-b055-e9d777fd8442.png">|<img width="273" alt="Screenshot 2022-10-07 at 19 22 28" src="https://user-images.githubusercontent.com/1182160/194615167-1b2874df-8f57-4821-8137-13029f18fd73.png">|

#### Testing Instructions

1. Use a test account that has more than one site.
2. Click on Stats (or Domains, Plugins, Posts, Pages).
3. Click on Switch Site.
4. Verify that the icons no longer appear.
5. Test in supported browsers.
6. Ensure that there are no other usages of the deleted component in the codebase.
7. Ensure that there are no other usages of `All Sites` where `sites` prop is passed.
8. Test in Calypso Green (Jetpack Cloud) locally with `CALYPSO_ENV=jetpack-cloud-development yarn run start`.

#### Pre-merge Checklist
